Sources/GRPCReflectionService/Documentation.docc/ReflectionServiceTutorial.md

```diff
@@ -0,0 +1,235 @@
+  ]
+
+let reflectionServiceProvider = try ReflectionService(
+  serializedFileDescriptorProtoFilePaths: paths, version: .v1
+)
+```
+
+### Running the Server
+
+In our example the server is not configured with TLS. The port is `1234`.
+Starting the server:
+
+```swift
+let server = try await Server.insecure(group: group)
+  .withServiceProviders([reflectionServiceProvider, GreeterProvider(), EchoProvider()])
+  .bind(host: "localhost", port: self.port)
+  .get()
+
+```
+
+To run the server, from the root of the package run:
+
+```sh
+$ swift run ReflectionServiceServer
+```
+
+ ## Testing the Reflection Service using GRPCurl
+
+Please follow the instructions from the [GRPCurl README][grpcurl-setup] in order to set gRPCurl up.
+
+From a different terminal than the one used for running the Server, we will call gRPCurl commands,
+following the format:
+
+`grpcurl [flags] [address] [list|describe] [symbol]`.
+
+In our case we are using the `-plaintext` flag, because our server isn't configured with TLS, and 
+the address is set to `localhost:1234`.
+
+Here are some gRPCurl commands and the responses:
+
+- **List services**
+```sh
+$ grpcurl -plaintext localhost:1234 list
+```
+
+**output:**
+```sh
+Echo
+Greeter
+ServerReflection
+```
+
+- **List methods of a service**
+```sh
+$ grpcurl -plaintext localhost:1234 list echo.Echo
+```
+
+**output:**
+```sh
+echo.Echo.Collect
+echo.Echo.Expand
+echo.Echo.Get
+echo.Echo.Update
+```
+
+- **Describe a service**
+```sh
+$ grpcurl -plaintext localhost:1234 describe echo.Echo
+```
+
+**output:**
+```
+echo.Echo is a service:
+service Echo {
+  // Collects a stream of messages and returns them concatenated when the caller closes.
+  rpc Collect ( stream .echo.EchoRequest ) returns ( .echo.EchoResponse );
+  // Splits a request into words and returns each word in a stream of messages.
+  rpc Expand ( .echo.EchoRequest ) returns ( stream .echo.EchoResponse );
+  // Immediately returns an echo of a request.
+  rpc Get ( .echo.EchoRequest ) returns ( .echo.EchoResponse );
+  // Streams back messages as they are received in an input stream.
+  rpc Update ( stream .echo.EchoRequest ) returns ( stream .echo.EchoResponse );
+}
+```
+
+- **Describe a method**
+```sh
+$ grpcurl -plaintext localhost:1234 describe echo.Echo.Collect
+```
+
+**output:**
+```
+echo.Echo.Collect is a method:
+// Collects a stream of messages and returns them concatenated when the caller closes.
+rpc Collect ( stream .echo.EchoRequest ) returns ( .echo.EchoResponse );
+```
+
+- **Describe a message type**
+```sh
+$ grpcurl -plaintext localhost:1234 describe echo.EchoRequest
+```
+
+**output:**
+```
+echo.EchoRequest is a message:
+message EchoRequest {
+  // The text of a message to be echoed.
+  string text = 1;
+}
+```
+- 
+
+-**Sending a request to the HelloWorld service**
+```sh
+$ grpcurl -d '{ "name": "World" }' -plaintext localhost:1234 helloworld.Greeter.SayHello
+```
+
+**output:**
+```
+{
+  "message": "Hello World!"
+}
+```
```

Comment:
Given the examples above use the Echo service I think it makes more sense to make a Get request on the Echo service.